featurebyte/api/feature_group.py

```diff
@@ -411,14 +418,69 @@ def _get_feature_list_batch_feature_create_payload(
                     tabular_source=feat.tabular_source,
                 )
             )
-        feature_list_create = FeatureListCreateWithBatchFeatureCreationPayload(
-            name=feature_list_name,
-            conflict_resolution=conflict_resolution,
+
+        # if feature with same name already exists, retrieve it
+        batch_feature_create = BatchFeatureCreatePayload(
             graph=pruned_graph,
             features=batch_feature_items,
+            conflict_resolution=conflict_resolution,
+        )
+        return batch_feature_create, batch_feature_items
+
+    @staticmethod
+    def _get_feature_list_create_with_batch_feature_creation_payload(
+        feature_list_id: ObjectId,
+        feature_list_name: str,
+        features: List[BatchFeatureItem],
+        conflict_resolution: ConflictResolution,
+    ) -> FeatureListCreateWithBatchFeatureCreationPayload:
+        return FeatureListCreateWithBatchFeatureCreationPayload(
             _id=feature_list_id,
+            name=feature_list_name,
+            features=features,
+            graph=QueryGraphModel(),
+            conflict_resolution=conflict_resolution,
+            skip_batch_feature_creation=True,
+        )
+
+    def _save_feature_list(
+        self,
+        feature_list_name: str,
+        feature_list_id: ObjectId,
+        conflict_resolution: ConflictResolution,
+    ) -> None:
+        # save features in batch
```

Comment:
Curious what is the rationale of saving in batches, and is there a reason to do the batching in the client vs in the server?

Reply:
I forgot to mention the main rationale. 😅 
When the number of feature is large, it causes error when saving the task to the mongo (since the feature cluster graph will be stored in the payload). That's why we limit the number of features per task to 500 to prevent this issue.

Reply:
I see. Then UI probably needs similar handling as well.

Reply:
Yes, we probably need to update feature list builder logic to prevent sending feature cluster graph.